core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -64,9 +68,12 @@ private[ui] class AllJobsPage(parent: JobsTab, store: AppStatusStore) extends We
     </svg></div>.toString.filter(_ != '\n')
 
   private def makeJobEvent(jobs: Seq[v1.JobData]): Seq[String] = {
+    val now = System.currentTimeMillis()
     jobs.filter { job =>
       job.status != JobExecutionStatus.UNKNOWN && job.submissionTime.isDefined
-    }.map { job =>
+    }.sortBy { j =>
+      (-j.completionTime.map(_.getTime).getOrElse(now), -j.submissionTime.get.getTime)
+    }.take(MAX_TIMELINE_JOBS).map { job =>
```

Comment:
nit: how about 
sortBy(completionTime, submissionTime).takeRight

the negative value is not straightforward.

Reply:
Thanks. It seems simpler.